misk/src/main/kotlin/misk/MiskServiceModule.kt

```diff
@@ -21,28 +21,35 @@ import misk.tokens.TokenGeneratorModule
  * with [MiskTestingServiceModule]. Only bindings that are not suitable for a unit testing
  * environment belong here.
  */
-class MiskRealServiceModule : KAbstractModule() {
+class MiskRealServiceModule(
+  private val serviceManagerConfig: ServiceManagerConfig,
+) : KAbstractModule() {
+  constructor(): this(ServiceManagerConfig())
+
   override fun configure() {
     install(ResourceLoaderModule())
     install(RealEnvVarModule())
     install(ClockModule())
     install(SleeperModule())
     install(TickerModule())
     install(TokenGeneratorModule())
-    install(MiskCommonServiceModule())
+    install(MiskCommonServiceModule(serviceManagerConfig))
   }
 }
 
 /**
  * This module has common bindings for all environments (both real and testing).
  */
-class MiskCommonServiceModule : KAbstractModule() {
+class MiskCommonServiceModule(
+  private val serviceManagerConfig: ServiceManagerConfig,
```

Comment:
Might want to have this include a default too instead of the secondary constructor? Or once it's rolled out internally, we should remove the secondary constructor.